### ISSUE TITLE
Cannot use Bearer Token in new asp.net core identity API

### ISSUE THREAD

**Issue by SandroRiz** [dotnet/AspNetCore.Docs#31727]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Following the sample published here:
https://github.com/dotnet/blazor-samples/tree/main/8.0/BlazorWebAssemblyStandaloneWithIdentity

And the documentation published here:
https://learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-identity?view=aspnetcore-8.0

I tried to setup the backend and the front end to use Bearer Token instead cookie but the WASM client doesn't login (but the Register endpoint works)

cc: @guardrex https://github.com/dotnet/AspNetCore.Docs/issues/31194 

### Expected Behavior

Change the Sample and /or the Documentation indicating 
- if the new asp.net core Identity API are suitable to be used eg. from a mobile client (that cannot use cookies)
- the risk of using bearer token
- if a WASM standalone client could use the token or only the cookies

### Steps To Reproduce

Change line 9 of backend program.cs adding .AddBearerToken() instead to .AddIdentityCookies() 
Change line 123 of frontend CookieAuthenticationStateProvider.cs putting false instead true

### .NET Version

8.0.0

### Anything else?

I was able to call from Postman the new API with bearer token NOT including the 

.AddAuthentication() but only the .AddAuthorization()

basically I copied what you obtain from the new Blazor 8 web app template when you choose to include identity... see attach

[Program.zip](https://github.com/dotnet/aspnetcore/files/13550181/Program.zip)

**Comment by guardrex** [comment]
My current understanding is that we aren't going to carry all of the code to implement this scenario per ...

> you then need to parse the `AccessTokenResponse` on the client and manage the access and refresh tokens yourself.

We've left language in the doc to that effect at ...

https://learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-identity?view=aspnetcore-8.0#token-authentication

@halter73 ... You can re-open this if you (and/or Jeremy) plan to provide additional guidance on this.

**Comment by lcetinapta** [comment]
Ok, how to parse the AccessTokenResponse.AccessToken?  In the .net 7.0 blazor wasm, I used JWT tokens, and they are easy to parse. But the Bearer token is not. First is encrypted, second it is not only base64 encoded but as i see in code it is in binary format. (I saw the code in BearerTokenOptions.Unprotect() method.
So I have the problem of using BearerTokens in my wasm blazor app where i need to create claims from it to be able to uze cleint side authorization on pages with the [Authorize] attribute.

**Comment by akordowski** [comment]
> Ok, how to parse the AccessTokenResponse.AccessToken?

@guardrex Can you tell if there is a way to parse the `AccessTokenResponse.AccessToken` in a Blazor Web App to obtain the claims in it? All articles and examples I could found only showing how ASP.NET Core Identity is used only in a web app, and not in a server/client scenario. Using the `_httpClient.PostAsJsonAsync("login?useCookies=true", ...)` approach in a server/client scenario, the login call was successful, but the user was still unauthorized in the web app. Any idea?

**Comment by akordowski** [comment]
@guardrex Thank you for your response! I have already read the articles, but they only cover the Identity integration in the BWA, not an external API.

In the meantime I found this **Standalone Blazor WebAssembly with ASP.NET Core Identity** [example](https://github.com/dotnet/blazor-samples/tree/main/9.0/BlazorWebAssemblyStandaloneWithIdentity), which uses an dedicated [`CookieAuthenticationStateProvider`](https://github.com/dotnet/blazor-samples/blob/main/9.0/BlazorWebAssemblyStandaloneWithIdentity/BlazorWasmAuth/Identity/CookieAuthenticationStateProvider.cs). In the [`GetAuthenticationStateAsync()`](https://github.com/dotnet/blazor-samples/blob/fc355e4056403422c284a64bd31b0f83252f2077/9.0/BlazorWebAssemblyStandaloneWithIdentity/BlazorWasmAuth/Identity/CookieAuthenticationStateProvider.cs#L161-L238) function it calls a custom endpoint on the API which returns the claims. I will test this example and will try to apply the approach to the token authentication.

> I'll try to reach that issue as soon as I can.

No worries, as for me it is only for a personal project, so no hurry. Thank you for reaching out!
